### DISCUSSION TITLE
Blocks

### DISCUSSION THREAD

**Discussion by azar-maghsoodi** [area17/twill#2240] [Q&A] [answered]
Hello

I create a block with this command php artisan twill:make:componentBlock name
and I did all the steps according to document, but my block is not shown in the form?
should I do register in appserviceprovider?

**Comment by ifox** [comment]
Did you include the block editor field to your form and specified which blocks would be available there?

**Reply by azar-maghsoodi** [reply]
yes I did , I write in the form
@php
        $blocks = [
            'gallery',
            'historicalLandmarks',
        ];
    @endphp
    <x-twill::block-editor
        :blocks="$blocks"
    />
    I have in the app/View/Components/Twill/Blocks/Gallery.php
    and in resources/views/components/twill/blocks/gallery.blade.php

**Reply by ifox** [reply]
If you run `artisan twill:list:blocks` you will see that component based blocks are named a bit differently. You can override `getName()` to change it, or use the generated name.

**Reply by azar-maghsoodi** [reply]
Thank you for the time
I run the command and I see the component is a17-block-app-gallery
but I did not catch you I have to change name of Gallery.php? and where I should override getName()?

**Reply by ifox** [reply, marked as answer]
In your array of blocks that you pass to the block editor field, instead of `gallery`, use the name value returned by the command, not the component value. 

Or, inside your `Gallery.php`, implement the `getName` method and return `gallery`.

**Reply by azar-maghsoodi** [reply]
Thank you very much. now it works!
